tiny_dnn/network.h

```diff
@@ -1035,6 +1046,7 @@ class network {
 
     std::string name_;
     NetType net_;
+    bool finish_training_;
```

Comment:
I personally think `stop_training_` or `halt_training_` or `interrupt_training_` is more suitable.